src/dmd/inline.d

```diff
@@ -1832,8 +1836,8 @@ private bool canInline(FuncDeclaration fd, bool hasthis, bool hdrscan, bool stat
     return true;
 
 Lno:
-    if (fd.inlining == PINLINE.always)
-        fd.error("cannot inline function");
+    if (fd.inlining == PINLINE.always && global.params.warnings == DiagnosticReporting.inform)
+        warning(fd.loc, "cannot inline function `%s`", fd.toChars());
```

Comment:
This no longer shows the fully qualified name, I'm not sure if there is a policy regarding that, but IMO the FQN should be kept for disambiguation.